src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -399,14 +399,14 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
                         else
                         {
                             // '{0}' cannot derive from special class '{1}'
-                            diagnostics.Add(ErrorCode.ERR_DeriveFromEnumOrValueType, Locations[0], this, baseType);
+                            diagnostics.Add(ErrorCode.ERR_DeriveFromEnumOrValueType, location, this, baseType);
                             continue;
                         }
                     }
 
                     if (baseType.IsSealed && !this.IsStatic) // Give precedence to ERR_StaticDerivedFromNonObject
                     {
-                        diagnostics.Add(ErrorCode.ERR_CantDeriveFromSealedType, Locations[0], this, baseType);
+                        diagnostics.Add(ErrorCode.ERR_CantDeriveFromSealedType, location, this, baseType);
```

Comment:
I made a compiler change to report the error on the base type instead of the deriving type declaration. I think this is better for a few reasons:
- It just seems better for clarity. If I see a squiggle on the base type, I immediately know where the problem is, but seeing a squiggle on the type declaration doesn't give me much of an idea as to what could be wrong. The problem is really with the base type, so why not make the location more specific to that?
- In case of a partial class, the current location might not even be at the same declaration. I don't think it's a good experience to show a squiggle for a problem that's in a completely different location.
- Other errors that have to do with invalid bases already report a location of the base itself, see below for `ERR_StaticDerivedFromNonObject`, `ERR_DuplicateInterfaceInBaseList` etc.
- This is what VB does as well.

Reply:
Tagging @jcouv @dotnet/roslyn-compiler

Reply:
I disagree with this reasoning, particularly 1. The problem is _not_ that the base class is sealed, the problem is that you cannot derive from a sealed class. This becomes problematic in several aspects:

1. `sealed` is a very intentional modifier. I tend to think of it as one of the most intentional modifiers, along with `private` and `readonly`, and certainly more intentional than deriving.
2. Our behavior for deriving from sealed classes from another compilation or dependency differs from the same compilation scenario.
3. The base class does not actually have a problem here. It is unaffected by someone trying to derive from it.

Reply:
I would rather see VB's behavior brought in line with C# tbh, as that feels like a bug to me.

Reply:
I think I worded my reasoning poorly. I agree the problem isn't that the base class is sealed - the problem is that I'm deriving from a sealed class. That's why I think the squiggle should be in the location where I'm specifying the base type - that's where I'm trying to derive from that class and that is the part that needs to be removed to fix the error.

Reply:
I misunderstood. For some reason I thought you meant squiggling the base class _definition_. I have no idea why I thought that was what you meant. Disregard my arguments, this sounds like a good change.

Reply:
@333fred Is it possible there was a misunderstanding? I am referring to the base specification, not the declaration of the base class.

Reply:
Yeah, this is what I get for looking at code reviews on my phone at midnight.

Reply:
@333fred I read it the same way at first and was equally confused: this statement:

> I made a compiler change to report the error on the base type instead of the deriving type declaration

It just says "base type". :smile:

Reply:
Sorry. I don't know what it's actually called, I haven't read the spec.